### DISCUSSION TITLE
Long running servers vs serverless

### DISCUSSION THREAD

**Discussion by zhex900** [epicweb-dev/epic-stack#418] [Q&A]
Hi,

Why was a long-running server chosen for this template? What about a serverless design? 

Here is my understanding:

Long-running: no cold start, fast response. Easy to set up a local development environment. The more complicated provisioning infra. 

Serverless: cold start. But can scale quickly. No cost when it is not used. Don't have to think about scaling.

**Comment by kentcdodds** [comment]
I need to write a decision document about this still, but Fly.io is a mix of long-running and serverless. You can have it scale up and down automatically :) https://fly.io/blog/the-serverless-server/

**Reply by gr2m** [reply]
Hey I was just looking for a decision document or anything else that explains why you decided for fly.io for deployment. This is my first day of looking into Remix 🎉 so while I have this outside perspective, a lack of guidance of where and how to deploy Remix with a list of tradeoffs is a bummer. The insight of long-running vs serverless is very interesting, thank you!
